### ISSUE TITLE
Sails 1.0: Helpers not being called (due to not executing the Deferred) / await is not defined on REPL  (because not inside of an `async function(){…}`)

### ISSUE THREAD

**Issue by alancornejo** [balderdashy/sails#4315]
**Sails version**: 1.0.0-46
**Node version**: 8.9.4
**NPM version**: 5.6.0 
**DB adapter name**: sails-mysql@beta
**DB adapter version**: N/A
**Operating system**: Windows 10 64 bits

<hr/>

What I need is a helper to call another helper.
When I try to make this call.

`sails sails.helpers.convertProyect('Claro',2)`

I throw the error that does not find sails, if I remove the await does not throw any error but it does not reach the second helper because I put a console.log and does not show me anything in the console to know that if I enter the second helper.

**Comment by sailsbot** [comment]
Hi @alancornejo!  It looks like you missed a step or two when you created your issue.  Please edit your comment (use the pencil icon at the top-right corner of the comment box) and fix the following:

* Verify "I am experiencing a concrete technical issue (aka a bug) with Sails (ideas and feature proposals should follow the guide for proposing features and enhancements (http://bit.ly/sails-feature-guide), which involves making a pull request).  If you're not 100% certain whether it's a bug or not, that's okay--you may continue.  The worst that can happen is that the issue will be closed and we'll point you in the right direction."
* Verify "I am not asking a question about how to use Sails or about whether or not Sails has a certain feature (please refer to the documentation(http://sailsjs.com), or post on http://stackoverflow.com, our Google Group (http://bit.ly/sails-google-group) or our live chat (https://gitter.im/balderdashy/sails)."
* Verify "I have already searched for related issues, and found none open (if you found a related _closed_ issue, please link to it in your post)."
* Verify "My issue title is concise, on-topic and polite ("jst.js being removed from layout.ejs on lift" is good; "templates dont work" or "why is sails dumb" are not so good)."
* Verify "I have tried all the following (if relevant) and my issue remains:"
* Verify "**I can provide steps to reproduce this issue that others can follow.**"

As soon as those items are rectified, post a new comment (e.g. &ldquo;Ok, fixed!&rdquo;) below and we'll take a look.  Thanks!

*If you feel this message is in error, or you want to debate the merits of my existence (sniffle), please contact inquiries@sailsjs.com

**Comment by sailsbot** [comment]
@alancornejo,@sailsbot: Hello, I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message and simply close the issue if possible.  On the other hand, if you are still waiting on a patch, please post a comment to keep the thread alive (with any new information you can provide).



If no further activity occurs on this thread within the next 3 days, the issue will automatically be closed.

Thanks so much for your help!

**Comment by mikermcneil** [comment]
@alancornejo this is because, without `await`, the 2nd helper is never being executed -- you're just building a Deferred.

I'm happy to report that, as of last week, if you reinstall your dependencies, you'll pick up a patch that should help debug this kind of stuff in the future.  At least when running in development mode, you'll see a warning log message about this after 15 seconds.  (See https://trello.com/c/7QnQZ6aC for more info.)

P.S. It looks like you're on the REPL (`sails console`) -- what I usually do (because you can't use `await` from there-- at least not yet) is just chain on `.log()` to the end.

**Comment by raqem** [comment]
Hello @alancornejo as Mike mentioned above this issue should now be resolved with a fresh publish of Sails. I am closing this issue in order to try and tidy-up the Sails issues.
